catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/WorkspaceTransformer.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer;
+
+import ddf.catalog.data.Metacard;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A utility used to convert workspace metacard data between the JSON data expected from frontend
+ * responses and the {@link Metacard} objects used in the backend.
+ *
+ * <p><b> This code is experimental. While this interface is functional and tested, it may change or
+ * be removed in a future version of the library. </b>
+ */
+public interface WorkspaceTransformer {
+  /**
+   * Transforms the given metacard JSON data according to available {@link WorkspaceTransformation}s
+   * and inserts the results into the given {@link Metacard} object.
+   *
+   * @param json the JSON-style map representing the metacard data
+   * @param init the {@link Metacard} into which the transformed data will be inserted
+   */
+  void transformIntoMetacard(Map<String, Object> json, Metacard init);
+
+  /**
+   * Transforms the given metacard JSON data into a new {@link Metacard} object.
+   *
+   * @param json the JSON-style map representing the metacard data
+   * @return a new {@link Metacard} containing the results of {@link WorkspaceTransformation}s of
+   *     the given JSON data
+   */
+  Metacard transform(Map<String, Object> json);
+
+  /**
+   * Transforms the given workspace {@link Metacard} into a JSON-style key-value pair {@link Map}
+   * using all available {@link WorkspaceTransformation}s.
+   *
+   * @param workspaceMetacard the workspace-type {@link Metacard} to be transformed
+   * @return a new JSON-style {@link Map} containing the results of {@link WorkspaceTransformation}s
+   *     performed on the given workspace {@link Metacard}'s data
+   */
+  default Map<String, Object> transform(Metacard workspaceMetacard) {
+    return transform(workspaceMetacard, workspaceMetacard);
+  }
+
+  /**
+   * Transforms the given {@link Metacard} into a JSON-style key-value pair {@link Map} using all
+   * available {@link WorkspaceTransformation}s.
+   *
+   * @param workspaceMetacard the workspace metacard that started the transformation; note that when
```

Comment:
What does "this will not be the same as the Metacard in the next argument" mean?

Also, does this javadoc mean that `metacard` has to be a metacard embedded in `workspaceMetacard`?

Reply:
It refers to the `metacard` argument directly below.